mne_bids/config.py

```diff
@@ -61,14 +61,30 @@
                            '.nwb',  # Neurodata without borders
                            ]
 
-ALLOWED_EXTENSIONS = {'meg': allowed_extensions_meg,
-                      'eeg': allowed_extensions_eeg,
-                      'ieeg': allowed_extensions_ieeg}
+ALLOWED_MODALITY_EXTENSIONS = {'meg': allowed_extensions_meg,
+                               'eeg': allowed_extensions_eeg,
+                               'ieeg': allowed_extensions_ieeg}
+
+# allowed kinds (i.e. last "_" delimiter in the BIDS filenames)
+ALLOWED_FILENAME_KINDS = [
+    'meg', 'markers', 'eeg', 'ieeg', 'T1w',  # modality
+    'participants', 'scans',
+    'electrodes', 'channels', 'coordsystem', 'events',  # sidecars
+    'headshape', 'digitizer',  # meg-specific sidecars
+    'behav', 'phsyio', 'stim'  # behavioral
+]
+
+# allowed BIDS extensions (extension in the BIDS filename)
+ALLOWED_FILENAME_EXTENSIONS = \
+    allowed_extensions_meg + \
+    allowed_extensions_eeg + \
+    allowed_extensions_ieeg + \
+    ['.json', '.tsv', '.tsv.gz', '.nii', '.nii.gz']
```

Comment:
Maybe we can get rid of those ugly backslashes:
```suggestion
ALLOWED_FILENAME_EXTENSIONS = (
    allowed_extensions_meg +
    allowed_extensions_eeg +
    allowed_extensions_ieeg +
    ['.json', '.tsv', '.tsv.gz', '.nii', '.nii.gz'])
```